agent/SharkserversBuyer.py

```diff
@@ -0,0 +1,192 @@
+from random import randint
+from selenium import webdriver
+from selenium.webdriver.common.desired_capabilities import DesiredCapabilities
+from selenium.webdriver.support.select import Select
+from selenium.webdriver.support.ui import WebDriverWait
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.common.exceptions import TimeoutException
+
+from BogusFormBuilder import BogusFormBuilder
+
+from VPSBuyer import VPSBuyer
+
+from Wallet import Wallet
+
+import selenium.webdriver.support.ui as ui
+
+import time
+
+from CountryGetter import CountryGetter
+
+
+class SharkserversBuyer(VPSBuyer):
+    '''
+    This class orders a VPS from sharkservers.co.uk
+    '''
+    def __init__(self, email = "", password = "", SSHPassword = ""):
+        super(SharkserversBuyer, self).__init__(email, password, SSHPassword)
+        self.email = email
+        if self.email == "":
+            self.email = self.generator.getEmail()
+            
+        if self.password != password or self.password == "":
+            self.password = self.generator.getRAString(32) + "!"
+            
+        if self.SSHPassword != SSHPassword or self.SSHPassword == "":
+            self.SSHPassword = self.generator.getRAString(32) # Generate new password
+            
+        self.SSHUsername = "root"
+        pass
+
+
+
+    def buy(self):
+        '''
+        Walks through the entire process of buying a VPS from Sharkservers. Returns True if it succeeded, returns False otherwise
+        '''
+        succeeded = self.placeOrder() # places the order
+        if succeeded == False:
+            return False
+
+        time.sleep(30) # Wait for half a minute so Sharkservers can process the payment
+
+        succeeded = self.getSSHInfo()
+        if succeeded == False:
+            return False
+        return True

```

Comment:
This is done for clarity's sake. If succeeded is false, it should quit the method and return false. Just like with the succeeded on line 48.

If it hasn't quit by then, the function has done everything with success, so it will return true.

What you proposed will have the same functionality, but to me it appears clearer in this way.